### ISSUE TITLE
Listener for disconnection of client

### ISSUE THREAD

**Issue by sclassen** [hivemq/hivemq-mqtt-client#264]
## Problem or use case
Application want's to be notified when the client gets disconnected from the broker.
This information is for example used to display status information in a UI

## Preferred solution or suggestions
I would favor something like the following

```
client.addDisconnectedListener(DisconnectedListener listener)

public interface DisconnectedListener {
    void handleDisconnect(DisconnectionReason reason);
}

public enum DisconnectionReason {
    CLIENT_DISCONNECTED,
    SERVER_DISCONNECTED,
    NETWORK_PROBLEM
}
```

Thanks @giftkugel for the cool hackaton

**Comment by SgtSilvio** [comment]
Thank you for your feedback.
This is already planned to be part of version 1.1.
Something similar will be added with the automatic reconnect handling (#83 )

**Comment by SgtSilvio** [comment]
This was closed as the DisconnectedListener was released as part of version 1.1
